ackhandler/sent_packet_handler.go

```diff
@@ -149,54 +138,58 @@ func (h *sentPacketHandler) ReceivedAck(ackFrame *frames.AckFrame, withPacketNum
 	if withPacketNumber <= h.largestReceivedPacketWithAck {
 		return ErrDuplicateOrOutOfOrderAck
 	}
-
 	h.largestReceivedPacketWithAck = withPacketNumber
 
 	// ignore repeated ACK (ACKs that don't have a higher LargestAcked than the last ACK)
 	if ackFrame.LargestAcked <= h.largestInOrderAcked() {
 		return nil
 	}
+	h.LargestAcked = ackFrame.LargestAcked
+
+	if h.skippedPacketsAcked(ackFrame) {
+		return ErrAckForSkippedPacket
+	}
 
-	// check if it acks any packets that were skipped
-	for _, p := range h.skippedPackets {
-		if ackFrame.AcksPacket(p) {
-			return ErrAckForSkippedPacket
+	rttUpdated := h.maybeUpdateRTT(ackFrame.LargestAcked, ackFrame.DelayTime, rcvTime)
+
+	ackedPackets, err := h.determineNewlyAckedPackets(ackFrame)
```

Comment:
True, this is slightly suboptimal. But I really like this structure, as it makes determineNewlyAckedPackets pure and therefore much easier to test (and isolates concerns between the functions better). I also want to try applying the same pattern to detectLostPackets, but it's a bit trickier to make that pure as well.